app/src/main/kotlin/com/intive/picover/parties/view/PartiesScreen.kt

```diff
@@ -56,8 +56,8 @@ fun PartiesScreen(
 		MVIState.Type.LOADING -> PicoverLoader(modifier = Modifier.fillMaxSize())
 		MVIState.Type.LOADED -> LoadedContent(
 			parties = state.parties,
-			onPartyClick = viewModel::onPartyClick,
-			onFabClick = viewModel::onAddPartyClick,
+			onPartyClick = { viewModel.emitEffect(NavigateToPartyDetails(it)) },
+			onFabClick = { viewModel.emitEffect(NavigateToAddParty) },
```

Comment:
probably we could get rid of all `onXXXClick` as they use the same method (viewModel.emitEffect), and have only one like `onEvent = viewModel::emitEffect`

Reply:
It's done in this way or missed something?

Reply:
I mean we have 
```
onPartyClick = { viewModel.emitEffect(NavigateToPartyDetails(it)) },
onFabClick = { viewModel.emitEffect(NavigateToAddParty) },
```
but it could be
```
onEvent = viewModel::emitEffect
```

and then
```
private fun LoadedContent(
	parties: List<Party>,
        onEvent: (PartiesEvent) -> Unit
) {
```

Reply:
Imagine case when `PartiesEvent` will be extended with additional events. Without any warning, it will allow to call `onEvent` callback with events that are not related to `LoadedContent` composable. Side effects are too general to pass them down in this case. It's the same reason why don't pass the view model down to each composable.

Reply:
I think passing the viewmodel is other case, we don't do it (or we shouldn't) cuz the tests.

I see your point, but to be honest, as it's only for us, we don't share the api for others, so the chance someone will call onEvent with wrong Event shouldn't happen.

Nevertheless it's only idea, I'm not forcing it :)

Reply:
I see your point as well and thank you for sharing it :) 

> as it's only for us, we don't share the api for others
In long term, such approach leads to hacky and unmaintainable code :)